online_editor/views.py

```diff
@@ -16,19 +16,46 @@ def index(request):
     context = {}
     return render(request, 'index.html', context)
 
-def execute(request):
-    code = request.body.decode() + "\n" #decoding byte to code
-    
+def run(request):
+    code = request.body.decode() + "\n"
+
     input_name = settings.INPUT_FILE_PATH
     output_name = settings.OUTPUT_FILE_PATH
     shell_script = settings.RUN_IN_DOCKER_SH_PATH
-    
+    if request.POST['compile_status'] != True:
+        response = {
+            "message": "The compilation is not complete yet"
+        }
+        return JsonResponse(response, safe=False)
+
+    run_data_frontend = {
+        'source': code,
+        'memory_limit': 243232,
+        'time_limit': 5,
+        'compile_status': True,
+        'lang': "python",
+        'input': "",
+        'id': "213121",
+    }
     # run in docker
-    result = run_in_docker(input_name, code, shell_script, output_name)
-    # enable pretty html formatting and obfuscate input file name with 'source'
-    result = result.replace('\n', "<br>").replace(input_name, "source")
+    try:
+        result = run_in_docker(input_name, code, shell_script, output_name)
+        # enable pretty html formatting and obfuscate input file name with 'source'
+        result = result.replace('\n', "<br>").replace(input_name, "source")
+    except:
+        pass
```

Comment:
Add some error-resolving code. For example, you can add a exact error message into your **_errors_** entity, for example, "Time_run_out" or "Memory_use_up".